src/libraries/System.Net.Connections/src/System/Net/Connections/DuplexPipeStream.cs

```diff
@@ -67,7 +67,7 @@ public override int Read(byte[] buffer, int offset, int count)
 
         public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            if (buffer == null) return Task.FromException<int>(ExceptionDispatchInfo.SetCurrentStackTrace(new ArgumentNullException(nameof(buffer))));
+            if (buffer == null) throw new ArgumentNullException(nameof(buffer));
```

Comment:
Why not jus call ValidateBufferArguments here?

Reply:
I can.  It'll result in some duplication with what the Memory ctor does, but not a big deal.